File: library/src/main/java/ru/terrakok/cicerone/android/support/SupportAppScreen.java
package ru.terrakok.cicerone.android.support;

import android.content.Context;
import android.content.Intent;
import androidx.fragment.app.Fragment;

import org.jetbrains.annotations.Nullable;

import ru.terrakok.cicerone.Screen;

/**
 * AppScreen is base class for description and creation application screen.<br>
 * NOTE: If you have described the creation of Intent then Activity will be started.<br>
 * Recommendation: Use Intents for launch external application.
 */
public abstract class SupportAppScreen extends Screen {

    public Fragment getFragment() {
        return null;
    }

    public Intent getActivityIntent(Context context) {
        return null;
    }

    @Nullable
    public FragmentParams getFragmentParams() {
        return null;
    }
}

File: library/src/main/java/ru/terrakok/cicerone/android/support/SupportAppScreen.kt
package ru.terrakok.cicerone.android.support

import android.content.Context
import android.content.Intent
import androidx.fragment.app.Fragment
import ru.terrakok.cicerone.Screen

abstract class SupportAppScreen : Screen() {

    open fun getFragment(): Fragment? = null

    open fun getActivityIntent(context: Context): Intent? = null

    open fun getFragmentParams(): FragmentParams? = null
}